cdk/lib/support-reminders.ts

```diff
@@ -1,15 +1,264 @@
+			monitoringConfiguration: {
+				snsTopicName: "conversion-dev",
+				http5xxAlarm: {
+					tolerated5xxPercentage: 1,
+				}
+			},
+			targets: [
+				{
+					path: "/reactivate",
+					httpMethod: "POST",
+					lambda: reactivateRecurringReminderLambda,
+				},
+				{
+					path: "/create/recurring",
+					httpMethod: "POST",
+					lambda: createRemindersSignupLambda,
+				},
+				{
+					path: "/search",
+					httpMethod: "POST",
+					lambda: searchRemindersLambda,
+				},
+				{
+					path: "/create/one-off",
+					httpMethod: "POST",
+					lambda: createRemindersSignupLambda,
+				},
+				{
+					path: "/cancel",
+					httpMethod: "POST",
+					lambda: cancelRemindersLambda,
+				},
+			],
+		})
+
+
+		// ---- Scheduled lambda functions ---- //
+		new GuScheduledLambda(this, "signup-exports", {
+			app,
+			runtime,
+			fileName,
+			vpc,
+			vpcSubnets,
+			securityGroups,
+			environment,
+			handler: "signup-exports/lambda/lambda.handler",
+			functionName: `support-reminders-signup-exports-${this.stage}`,
+			rules: [
+				{
+					schedule: Schedule.expression("cron(05 00 * * ? *)"),
+				},
+			],
+			monitoringConfiguration: {
+				snsTopicName: "conversion-dev",
+				toleratedErrorPercentage: 1,
+			},
+		});
+
+		new GuScheduledLambda(this, "next-reminders", {
+			app,
+			runtime,
+			fileName,
+			vpc,
+			vpcSubnets,
+			securityGroups,
+			environment,
+			handler: "next-reminders/lambda/lambda.handler",
+			functionName: `support-reminders-next-reminders-${this.stage}`,
+			rules: [
+				{
+					schedule: Schedule.expression("cron(05 00 * * ? *)"),
+				},
+			],
+			monitoringConfiguration: {
+				snsTopicName: "conversion-dev",
+				toleratedErrorPercentage: 1,
+			},
+		});
+
+
+		// ---- DNS ---- //
+		const certificateArn = `arn:aws:acm:eu-west-1:${this.account}:certificate/${props.certificateId}`;
+
+		const cfnDomainName = new CfnDomainName(this, "DomainName", {
+			domainName: props.domainName,
+			regionalCertificateArn: certificateArn,
+			endpointConfiguration: {
+				types: ["REGIONAL"]
+			}
+		});
+
+		new CfnBasePathMapping(this, "ApiMapping", {
+			domainName: cfnDomainName.ref,
+			restApiId: supportRemindersApi.api.restApiId,
+			stage: supportRemindersApi.api.deploymentStage.stageName,
+		});
+
+		new CfnRecordSet(this, "DNSRecord", {
+			name: props.domainName,
+			type: "CNAME",
+			hostedZoneId: props.hostedZoneId,
+			ttl: "60",
+			resourceRecords: [
+				cfnDomainName.attrRegionalDomainName
+			],
+		});
+
+
+		// ---- Apply policies ---- //
+		const ssmInlinePolicy: Policy = new Policy(this, "SSM inline policy", {
+			statements: [
+				new PolicyStatement({
+					effect: Effect.ALLOW,
+					actions: [
+						"ssm:GetParametersByPath",
+						"ssm:GetParameter"
+					],
+					resources: [
+						`arn:aws:ssm:${Aws.REGION}:${Aws.ACCOUNT_ID}:parameter/support-reminders/db-config/${props.stage}`,
+						`arn:aws:ssm:${Aws.REGION}:${Aws.ACCOUNT_ID}:parameter/support-reminders/idapi/${props.stage}/*`,
```

Comment:
These paths for config are unconventional. The convention is to place config on `/STAGE/STACK/APP/config-name`. Let's update in a future PR?

Reply:
The region and account ID are also properties of `this` (an instance of `GuStack`), so we can simplify this slightly:

```ts
`arn:aws:ssm:${this.region}:${this.account}:parameter/support-reminders/db-config/${props.stage}`,
```